### ISSUE TITLE
Trying to run the code in dcgan_demo.py but the samples are not created in the 'real' folder of training set.

### ISSUE THREAD

**Issue by ChaoShen0** [xuqiantong/GAN-Metrics#1]
![image](https://user-images.githubusercontent.com/44055265/47521418-e393c980-d88a-11e8-996e-bc386a392398.png)
![image](https://user-images.githubusercontent.com/44055265/47521462-f908f380-d88a-11e8-96aa-43ac2bd47794.png)
![image](https://user-images.githubusercontent.com/44055265/47521514-248bde00-d88b-11e8-87be-f46df13a9172.png)

Apparently no real sample is generated in the target training folder when the code is running in training sessions. Please check it and possibly fix it.

**Comment by youyuge34** [comment]
The bugs are easy to be fixed by following:  
- If  windows, make the codes in `demo_dcgan.py` run under 
`
if __name__ == '__main__':
`  to handle with the multi-process errors.      
(or make the parameter `num_worker`  to 0 in all `dataloader()`   functions)    
- open `metric.py` file in the root directory, line 105-116 which should be     

```
    iter = 0
    for i, data in enumerate(dataloader, 0):
        img, _ = data
        for j in range(0, len(img)):

            vutils.save_image(img[j].mul(0.5).add(
                0.5), saveFolder + giveName(iter) + ".png")
            iter += 1
            if iter >= sampleSize:
                break
        if iter >= sampleSize:
            break
```

the author input redundant `tab`s to these lines. use `shift+tab` to delete the tabs ! That's why no real sample is generated in the folder.

**Comment by xuqiantong** [comment]
Thanks @youyuge34 for pointing out the problem! Would you mind to send out a PR to fix it?

**Comment by youyuge34** [comment]
@xuqiantong I have made a PR which fixed several bugs and modified the structure. Details are shown in PR comment.     
THX for your paper and codes again.

**Comment by xuqiantong** [comment]
Bug fixed by @youyuge34 in https://github.com/xuqiantong/GAN-Metrics/pull/2. Closing.
